RELEASE.md

```diff
@@ -14,12 +14,45 @@
     * Limitador version the operator enables installations of (without prefix) – i.e. `0.X.Y`
     * If the release is a prerelease
 
-3. Verify that the build [release tag workflow](https://github.com/Kuadrant/dns-operator/actions/workflows/build-images-for-tag-release.yaml) is triggered and completes for the new tag
+3. Verify that the build [release tag workflow](https://github.com/Kuadrant/limitador-operator/actions/workflows/build-images-for-tag-release.yaml) is triggered and completes for the new tag
 
 4. Verify the new version can be installed from the catalog image, see [Verify OLM Deployment](#verify-olm-deployment)
 
 5. Release to the [community operator index catalogs](#community-operator-index-catalogs).
 
+### Verify OLM Deployment
+
+1. Deploy the OLM catalog image following the [Deploy the operator using OLM](/doc/development.md#deploy-the-operator-using-olm) and providing the generated catalog image. For example:
+```sh
+make deploy-catalog CATALOG_IMG=quay.io/kuadrant/limitador-operator-catalog:v0.13.0
+```
+
+2. Wait for deployment:
+```sh
+kubectl -n limitador-system wait --timeout=60s --for=condition=Available deployments --all
+```
+
+The output should be:
+
+```
+deployment.apps/limitador-operator-controller-manager condition met
+```
+
+3. Check the logs:
+```sh
+kubectl -n limitador-system logs -f deployment/limitador-operator-controller-manager
+```
+
+4. Check the version of the components deployed:
+```sh
+kubectl -n limitador-system get deployment -o yaml | grep "image:"
+```
+The output should be something like:
+
+```
+image: quay.io/kuadrant/limitador-operator:v0.13.0
+```
+
 ### Community Operator Index Catalogs
```

Comment:
Is this section still relevant?

Reply:
No, I would remove it and also from other kuadrant component release docs